lib/scholar/metrics.ex

```diff
@@ -331,6 +331,76 @@ defmodule Scholar.Metrics do
     Nx.divide(true_negatives, false_positives + true_negatives + 1.0e-16)
   end
 
+  @doc ~S"""
+  Calculates the confusion matrix.
+
+  ## Argument Shapes
+
+    * `y_true` - $\(d_0, d_1, ..., d_n\)$
+    * `y_pred` - $\(d_0, d_1, ..., d_n\)$
+
+  ## Examples
+
+      iex> y_true = Nx.tensor([1.0, 1.0, 2.0, 2.0, 3.0, 3.0], type: {:f, 32})
+      iex> y_pred = Nx.tensor([1.0, 2.0, 1.0, 3.0, 3.0, 3.0], type: {:f, 32})
+      iex> Scholar.Metrics.confusion_matrix(y_true, y_pred)
+  """
+  def confusion_matrix(y_true, y_pred) do
```

Comment:
We want to use `defn` whenever possible. I think it makes sense to require `y_true` and `y_pred` to be integer tensors, with class numbers starting from 0. Then we need a required option like `:num_classes`, because we need to know the confusion matrix size upfront.

Given that, one way to implement this would be creating a zeroed confusion matrix, then then concatenate `y_true` and `y_pred` to create indices into the matrix, and finally do `indexed_add` with those indices and ones as values.

Related: [tf.math.confusion_matrix](https://www.tensorflow.org/api_docs/python/tf/math/confusion_matrix), [sklearn.metrics.confusion_matrix](https://scikit-learn.org/stable/modules/generated/sklearn.metrics.confusion_matrix.html).

Reply:
Cool! But I'm not sure about how to create a zero tensor with a specific shape. Is there any function to achieve that, like `Nx.zeros/1` or `Nx.ones/1`?

Reply:
You want Nx.broadcast(0, shape)